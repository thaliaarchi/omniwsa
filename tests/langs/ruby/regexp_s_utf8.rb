puts "Ruby #{RUBY_VERSION}"
puts 'UTF-8 whitespace characters for Regexp \s:'
pattern = /^\s$/
(0..0x10FFFF).each do |c|
  next if c.between?(0xD800, 0xDFFF)
  s = c.chr(Encoding::UTF_8)
  if s =~ pattern
    puts "U+#{c.to_s(16).upcase.rjust(4, '0')} (#{s.inspect})"
  end
end
