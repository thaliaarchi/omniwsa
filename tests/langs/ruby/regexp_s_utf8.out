Ruby 3.3.6
UTF-8 whitespace characters for Regexp \s:
U+0009 ("\t")
U+000A ("\n")
U+000B ("\v")
U+000C ("\f")
U+000D ("\r")
U+0020 (" ")
